File: frontend/src/lib/types/order.ts
export interface BookAtPurchase {
	book_id: string;
	title: string;
	genre: string;
	img_path: string;
	quantity: number;
	price_at_purchase: number;
}

export interface OrderDetail {
	id: string;
	total_price: number;
	order_status: string;
	created_at: string;
	updated_at: Date;
	items: BookAtPurchase[];
}

File: backend/src/schemas/order.rs
use chrono::{DateTime, Utc};
use rust_decimal::Decimal;
use serde::Serialize;
use uuid::Uuid;

// Response Schema
#[derive(Serialize, Clone)]
pub struct BookAtPurchase {
   pub book_id: Uuid,
   pub title: String,
   pub genre: String,
   pub img_path: String,
   pub quantity: i32,
   pub price_at_purchase: Decimal,
}

#[derive(Serialize)]
pub struct OrderDetail {
   pub id: Uuid,
   pub total_price: Decimal,
   pub order_status: String,
   pub created_at: DateTime<Utc>,
   pub updated_at: DateTime<Utc>,
   pub items: Vec<BookAtPurchase>,
}

// For admin
#[derive(Serialize)]
pub struct Order {
   pub id: Uuid,
   pub user_id: Uuid,
   pub total_price: Decimal,
   pub order_status: String,
   pub created_at: DateTime<Utc>,
   pub updated_at: DateTime<Utc>,
}
